### PR TITLE
[helm] Add configuration vars for liveness and readiness probes

### PR DESCRIPTION
**Pull request by bykvaadm**
closes issue #62

### CODE REVIEW

**Comment by brotandgames** [comment]
@bykvaadm Thanks for your PR!

Could we refactor one thing:

Replace
````
liveness: true
livenessProbe:
  initialDelaySeconds: 60
````
with
````
livenessProbe:
  enabled: true
  initialDelaySeconds: 60
````

Also for readinessProbe.

Following best practices from the Kubernetes community: https://github.com/helm/charts/blob/master/stable/artifactory/values.yaml#L87

**Comment by brotandgames** [comment]
@bykvaadm We don't want to put variables out of the env.* scope in values.yaml.

Check https://github.com/brotandgames/ciao#configuration

It would be enough to check if `env.BASIC_AUTH_USERNAME` is set to decide if liveness or readiness Probe should use it.

**Comment by bykvaadm** [comment]
u mean to check this?

```
          livenessProbe:
            httpGet:
              path: /
              port: http
              {{- if .Values.env.BASIC_AUTH_USERNAME }}
...
```
and remove basic_auth section at all

File: chart/templates/deployment.yaml [hunk 1]
```diff
@@ -46,6 +52,11 @@ spec:
             httpGet:
               path: /
               port: http
+              {{- if .Values.basic_auth.enabled }}
```
Comment (brotandgames) [inline chart/templates/deployment.yaml]: @bykvaadm 

Use the following instead:
````
  {{- if .Values.env.BASIC_AUTH_USERNAME }}
  
  {{- end }}
````

in both probes and remove the if-block above and the additions in values.yaml.

Comment (brotandgames) [reply chart/templates/deployment.yaml]: That way it remains KISS :)

Comment (bykvaadm) [reply chart/templates/deployment.yaml]: the if-block above must remain so user can disable probes if they wish

**Comment by bykvaadm** [comment]
done

File: chart/templates/deployment.yaml (hunk 1 above)
Comment (brotandgames) [reply chart/templates/deployment.yaml]: if-block above =
````
            {{- if .Values.basic_auth.enabled }}
            - name: BASIC_AUTH_USERNAME
              value: "{{ .Values.basic_auth.username }}"
            - name: BASIC_AUTH_PASSWORD
              value: "{{ .Values.basic_auth.password }}"
            {{- end }}
````

File: chart/values.yaml [hunk 2]
```diff
@@ -21,6 +21,8 @@ fullnameOverride: ""
 # All ciao configuration options are adjustable via ENV vars.
 # Ref: https://github.com/brotandgames/ciao#configuration
 env: {}
+  # BASIC_AUTH_USERNAME: "ciao-admin"
```
Comment (brotandgames) [inline chart/values.yaml]:
```suggestion
```

File: chart/values.yaml [hunk 3]
```diff
@@ -21,6 +21,8 @@ fullnameOverride: ""
...
+  # BASIC_AUTH_PASSWORD: "mysecreto"
```
Comment (brotandgames) [inline chart/values.yaml]:
```suggestion
```
This is clear from the ENV examples.

**Comment by brotandgames** [comment]
@bykvaadm Thanks for your work!

**Comment by bykvaadm** [comment]
why don't u want to keep commented basic auth vars in values?

**Comment by brotandgames** [comment]
Because they are already commented in the README.md and we want to keep only a set of ENV variables consistent among configuration options.

**Pull request merged**